### ISSUE TITLE
Return type of function incorrectly resolved

### ISSUE THREAD

**Issue by overlookmotel** [oxc-project/oxc#6387]
```ts
export type T = number;

function foo(n: number): T {
  type T = string;
  return n;
}
```

`T` in return type of `foo` should resolve to top level `export type T`, but instead gets resolved to `type T` inside the function body. [Oxc Playground](https://playground.oxc.rs/#eNpFTjtuwzAMvYrAqQWMHkBF18wdOnqRXSoQIJECSaUxDN+9chrXE/n4flxhBg94ryzmbKnovtyHo1YmlPeRRoqNZktMLjK/kH9Sr77r1pHc6VGTRNfucU7QmpCjDraRYAAGv4I02ocuZOEO3qThADmRgY8hawc6c8WD0aVMnA9kEkgjS3mKtwFqEEV5RHKTGS8pI4XSA8BQ7c20N1fBLrrhZxAkPc1778Pc9//s4zDzN15x/7aDkijFdFJkwvmS+WdvvqFMrL3yL3fbfgG1h3ai)

TypeScript resolves it to the top-level `T`. [TS Playground](https://www.typescriptlang.org/play/?#code/KYDwDg9gTgLgBDAnmYcAqcC8cB2BXAWwCNgoBuAKAoDM8cBjGASwhzmoggAocAuXQiSgBKfhgDeFOAmSoM2AM4woTHAHNK0qMBh4obHJQC+FIA). If it was resolving to `type T`, the code would fail type check.

**Comment by Boshen** [comment]
@Dunqing can you help and resolve this once and for all 😅

**Comment by overlookmotel** [comment]
I think #6388 solves this. It just needs tests.
